desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -96,22 +96,33 @@ def handle_on_message(channel_id, bot_id, elements, user_id):
       _send_message(channel_id, bot_message)
 
     if text.lower().startswith('select'):
-      detect_select_statement(channel_id, user_id, text.lower())
-
-
-def detect_select_statement(channel_id, user_id, statement):
-  if not statement.endswith(';'):
-    statement += ';'
-  
-  raw_sql_message = 'Hi <@{user}>\nLooks like you are copy/pasting SQL, instead now you can send Editor links which unfurls in a rich preview!'.format(user=user_id)
-  _send_ephemeral_message(channel_id, user_id, raw_sql_message)
-
+      detect_select_statement(host_domain, is_http_secure, channel_id, user_id, text.lower())
+
+
+def detect_select_statement(host_domain, is_http_secure, channel_id, user_id, statement):
+  slack_user = check_slack_user_permission(host_domain, user_id)
+  user = get_user(channel_id, slack_user)
+
+  gist_doc = Document2.objects.create(
+    name='Mysql Query blah',
+    type='gist',
+    owner=user,
+    data=json.dumps({'statement_raw': statement}),
+    extra='mysql',
+    parent_directory=Document2.objects.get_gist_directory(user)
+  )
+
+  gist_link = '%(scheme)s://%(host)s/hue/gist?uuid=%(uuid)s' % {
+    'scheme': 'https' if is_http_secure else 'http',
+    'host': host_domain,
+    'uuid': gist_doc.uuid,
+  }
```

Comment:
Isn't this a duplication of the gist creation functionality?

Reply:
Yes, however gist_create is extracting required params (name, statement, etc) from the request argument @romainr
Also, to create link it takes host and scheme from request

Is there a need to create a helper function for gist_create  or am I missing something @romainr ??

Reply:
Yes missing that we are duplicating create gist functionality?

Reply:
Created a helper function for gist creation to remove duplication